many: make save/load hooks mandatory if affecting ephemeral

miguelpires opened the pull request:
Make save-view/load-view hooks mandatory only if the operation might affect ephemeral data. If the path traverses or is a parent of a subpath with ephemeral data, then the save/load hooks must be defined. Based on https://github.com/canonical/snapd/pull/15349

andrewphelpsj commented on confdb/confdb.go:
```diff
@@ -1063,6 +1063,99 @@ func mergeNamespaces(old, new interface{}) (interface{}, error) {
+	for i, request := range requests {
+		reqMatches, err := v.matchGetRequest(request)
+		if err != nil {
+			if errors.Is(err, &NotFoundError{}) && i != len(requests)-1 {
```
Rather than returning the last `NotFoundError`, should this return a different, more specific error, after the for loop, if `len(matches) == 0`?

andrewphelpsj commented on confdb/confdb.go:
```diff
@@ -1063,6 +1063,99 @@ func mergeNamespaces(old, new interface{}) (interface{}, error) {
+		}
+
+		if len(pathParts) == 0 {
+			if eph := schema.NestedEphemeral(); eph {
```
nitpick: `eph` isn't needed here

andrewphelpsj reviewed (commented):
Thanks! This LGTM, but will wait to approve until prereqs are merged.

miguelpires commented:
@andrewphelpsj @ZeyadYasser I've rebased this on master to pull in the changes from the other PR and it's ready for review (there were no other changes since the last push)

ZeyadYasser reviewed (approved):
Thank you!
